second deliverable

juangrind opened the pull request:
- Room
- OkHtt -lambdas and extension function
- unit tests
- ConstraintLayout

Commit:
second deliverable
- Room
- OkHtt
 -lambdas and extension function
- unit tests
- ConstraintLayout

mrarrieta commented on app/src/test/java/com/jpgl/cryptocurrencies/domain/GetAvailableBookUseCaseTest.kt:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    @Test
+    fun `Si hay conexion a internet y getAllAvailableBooksFromApi regresa una lista con datos`() = runBlocking {
```
esta no es la convencion de google para nombrar pruebas unitarias.

mrarrieta commented on app/src/test/java/com/jpgl/cryptocurrencies/domain/CryptoRepositoryTest.kt:
```diff
@@ -0,0 +1,42 @@
+    fun `test api`() = runBlocking {
+        coEvery { cryptoRepository.getAllAvailableBooksFromApi() } returns emptyList<BooksModelDomain>()
+        var myList = emptyList<BooksModelDomain>()
+
```
aca no veo el given when y then

mrarrieta commented on app/src/main/res/layout/fragment_crypto_detail.xml:
```diff
@@ -1,116 +1,120 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    tools:context=".ui.view.CryptoDetailFragment">
+
+    <LinearLayout
```
deberias usar constraint para todo

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/ui/viewModel/CryptoViewModel.kt:
```diff
@@ -54,6 +55,7 @@ class CryptoViewModel @Inject constructor(
         }
     }
 
+    @SuppressLint("SuspiciousIndentation")
```
la solucion es indentar bien, no quitar la revision

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/ui/view/MainActivity.kt:
```diff
@@ -27,6 +22,29 @@ class MainActivity : AppCompatActivity() {
         setContentView(binding.root)
 
+        cryptoViewModel.onCreateAvailableBook()
+        cryptoViewModel.bookState.observe(this) {
```
creo que estos errores los deberia manejar cada vista.

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/domain/GetBidsUseCase.kt:
```diff
@@ -6,13 +6,14 @@ import com.jpgl.cryptocurrencies.data.model.BidsModel
 ) {
     suspend operator fun invoke(book: String): List<BidsModelDomain> {
-        val bids =  cryptoRepository.getAllBidsFromApi(book)
+        val bids = if(BaseUtils.isNetworkEnabled()) cryptoRepository.getAllBidsFromApi(book) else cryptoRepository.getAllBidsFromDatabase()
```
esto lo deberia hacer el repo

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/domain/GetAvailableBookUseCase.kt:
```diff
@@ -2,22 +2,26 @@ package com.jpgl.cryptocurrencies.domain
+) {
     suspend operator fun invoke(): List<BooksModelDomain> {
-        val books =  cryptoRepository.getAllAvailableBooksFromApi()
+        val books = if (BaseUtils.isNetworkEnabled()) {
```
esto lo deberia hacer el repo

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/domain/GetAsksUseCase.kt:
```diff
@@ -1,23 +1,23 @@
 ) {
     suspend operator fun invoke(book: String): List<AsksModelDomain> {
-        val asks =  cryptoRepository.getAllAsksFromApi(book)
+        val asks = if (BaseUtils.isNetworkEnabled()) cryptoRepository.getAllAsksFromApi(book) else cryptoRepository.getAllAsksFromDatabase()
```
igual aca

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/data/database/entities/AskEnity.kt:
```diff
@@ -6,13 +6,14 @@ import androidx.room.PrimaryKey
-data class AsksEntity (
+data class AsksEntity(
     @PrimaryKey(autoGenerate = true)
-    @ColumnInfo(name = "idAsks") val idAsks: Int = 0,
```
si el nombre es el mismo no necesitas ColumnInfo

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/data/database/entities/BidsEntity.kt:
```diff
@@ -1,21 +1,19 @@
 @Entity(tableName = "table_bids")
 data class BidsEntity(
     @PrimaryKey(autoGenerate = true)
-    @ColumnInfo(name = "idBid") val idBid: Int = 0,
```
aca igual

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/data/database/entities/BookEntity.kt:
```diff
@@ -3,20 +3,20 @@ package com.jpgl.cryptocurrencies.data.database.entities
 @Entity(tableName = "table_book")
 data class BookEntity(
     @PrimaryKey(autoGenerate = true)
-    @ColumnInfo(name = "idBook") val idBook: Int = 0,
```
y aca

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/domain/GetTickerUseCase.kt:
```diff
@@ -1,26 +1,23 @@
 
     suspend operator fun invoke(book: String): TickerModelDomain? {
-        val ticker =  cryptoRepository.getAllTickerFromApi(book)
+        val ticker = if (BaseUtils.isNetworkEnabled()) cryptoRepository.getAllTickerFromApi(book) else cryptoRepository.getAllTickerFromDatabase()
```
esto lo debe decidir el repo

mrarrieta commented on app/src/main/java/com/jpgl/cryptocurrencies/utils/BaseUtils.kt:
```diff
@@ -3,12 +3,33 @@ package com.jpgl.cryptocurrencies.utils
             field = value
         }
 
+    fun isNetworkEnabled(): Boolean {
```
esto no deberia ser un objeto, las herramientas de logica deben ser inyectables y testeados.

mrarrieta reviewed (commented):
la carpeta de .idea y .gradle no deberian ir en el repo.

Commit:
Third deliverable
-linter.
-Inline functions, tail recursive, delegated properties and collection operations
-Coroutines and suspend functions.
-Hilt  dependencies inyection
-Navigation componen
-RxJava2

juangrind commented:
Second deliverable
